spec/predicates/release.md

```diff
@@ -0,0 +1,95 @@
+# Predicate type: Release
+
+Type URI: https://in-toto.io/attestation/release
+
+Version 0.1
+
+## Purpose
+
+To authoritatively link from a specific release name and version string in a registry, to the artifact names and hashes that make up that release.
+
+## Use Cases
+
+When receiving a new release version, package registries can publish a release attestation covering the artifact names and hashes that make up that release. This allows consumers of that release version to ensure the artifacts they are consuming have not been tampered with.
+
+If these release attestations are optionally published to a transparency log, package authors (or other interested parties) can monitor when a new version of a package is released.
+
+These use cases are not hypothetical; both of them are the case today for npm's [build provenance feature], which includes a [publish attestation]. Note that while npm calls this a publish attestation, calling it a release attestation better reflects that it's coming from the package registry. Publishing often refers to an author sending content to the registry, as in [PyPI's trusted publishers feature].
+
+## Prerequisites
+
+This predicate depends on the [in-toto attestation framework], as well as the [purl-spec] for identifying packaging ecosystems.
+
+Perhaps surprisingly, this predicate does not depend on [SLSA Provenance], but they are better together. In an ecosystem where some (but not all) packages have SLSA Provenance and you query by artifact hash, you can't tell the difference between an artifact that has been tampered with and one that does not yet have SLSA provenance. If the ecosystem has release attestations, you can give authoritative answers to what artifacts make up a given release version, and what hashes those artifacts should have.
+
+## Model
+
+This predicate is for the final stages of the software supply chain, where consumers are looking for attestations that the software has not been tampered with during distribution.
+
+## Schema
+
+### Fields
+
+- **purl, required** string (ResourceURI)
+  - A purl uniquely identifying a specific release name and version from a package registry.
```

Comment:
Should this be as specific as a complete URL with a domain name then? I would imagine so.

Reply:
Ideally this field would uniquely identify a release name and version. purl isn't exactly this, but if you make the version string required (which is optional in the purl spec) and leave off the qualifiers / subpath, I think it can uniquely identify a release name and version.

URLs are a bit too free-form: should you point to the README? The download page? It's very easy to construct multiple distinct URLs that point to the same resource.

The downside of using purls is that ecosystems would need to be accepted as a [known purl type](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst) which could be seen as a barrier to entry. But I think there's also value in this formalization.

Reply:
Should you state here that this MUST include a version, SHOULD include a qualifier if and only if it's necessary to scope a package name (as is the case for `oci`, for example), SHOULD NOT include a subpath?

Reply:
Good suggestion - I've updated the Parsing Rules below to be more explicit.